// @cds/angular/esm2015/cds/components/navigation/cds-navigation-start.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsNavigationStartDirective extends BaseCdsDirective {
    get i18n() {
        return this.element['i18n'];
    }
    set i18n(value) {
        this.element['i18n'] = value;
    }
    get expandedRoot() {
        return this.element['expandedRoot'];
    }
    set expandedRoot(value) {
        if (value === '') {
            this.element['expandedRoot'] = true;
            return;
        }
        this.element['expandedRoot'] = value;
    }
    get hasFocus() {
        return this.element['hasFocus'];
    }
    set hasFocus(value) {
        this.element['hasFocus'] = value;
    }
    get isGroupStart() {
        return this.element['isGroupStart'];
    }
    set isGroupStart(value) {
        if (value === '') {
            this.element['isGroupStart'] = true;
            return;
        }
        this.element['isGroupStart'] = value;
    }
    get role() {
        return this.element['role'];
    }
    set role(value) {
        this.element['role'] = value;
    }
    get navigationGroupId() {
        return this.element['navigationGroupId'];
    }
    set navigationGroupId(value) {
        this.element['navigationGroupId'] = value;
    }
    get itemText() {
        return this.element['itemText'];
    }
    set itemText(value) {
        this.element['itemText'] = value;
    }
    get toggleIconDirection() {
        return this.element['toggleIconDirection'];
    }
    set toggleIconDirection(value) {
        this.element['toggleIconDirection'] = value;
    }
}
CdsNavigationStartDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-navigation-start' },] }
];
CdsNavigationStartDirective.propDecorators = {
    i18n: [{ type: Input }],
    expandedRoot: [{ type: Input }],
    hasFocus: [{ type: Input }],
    isGroupStart: [{ type: Input }],
    role: [{ type: Input }],
    navigationGroupId: [{ type: Input }],
    itemText: [{ type: Input }],
    toggleIconDirection: [{ type: Input }]
};
