// @cds/angular/esm2015/cds/components/navigation/cds-navigation.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input, Output, EventEmitter } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsNavigationDirective extends BaseCdsDirective {
    constructor() {
        super(...arguments);
        this.cdsMotionChange = new EventEmitter();
    }
    get expandedRoot() {
        return this.element['expandedRoot'];
    }
    set expandedRoot(value) {
        if (value === '') {
            this.element['expandedRoot'] = true;
            return;
        }
        this.element['expandedRoot'] = value;
    }
    get cdsMotion() {
        return this.element['cdsMotion'];
    }
    set cdsMotion(value) {
        this.element['cdsMotion'] = value;
    }
    get role() {
        return this.element['role'];
    }
    set role(value) {
        this.element['role'] = value;
    }
    get ariaActiveDescendant() {
        return this.element['ariaActiveDescendant'];
    }
    set ariaActiveDescendant(value) {
        this.element['ariaActiveDescendant'] = value;
    }
    get expanded() {
        return this.element['expanded'];
    }
    set expanded(value) {
        if (value === '') {
            this.element['expanded'] = true;
            return;
        }
        this.element['expanded'] = value;
    }
    get i18n() {
        return this.element['i18n'];
    }
    set i18n(value) {
        this.element['i18n'] = value;
    }
    get currentActiveItem() {
        return this.element['currentActiveItem'];
    }
    set currentActiveItem(value) {
        this.element['currentActiveItem'] = value;
    }
}
CdsNavigationDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-navigation' },] }
];
CdsNavigationDirective.propDecorators = {
    expandedRoot: [{ type: Input }],
    cdsMotion: [{ type: Input }],
    role: [{ type: Input }],
    ariaActiveDescendant: [{ type: Input }],
    expanded: [{ type: Input }],
    i18n: [{ type: Input }],
    currentActiveItem: [{ type: Input }],
    cdsMotionChange: [{ type: Output }]
};
