// @cds/angular/esm2015/cds/components/navigation/cds-navigation-group.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input, Output, EventEmitter } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsNavigationGroupDirective extends BaseCdsDirective {
    constructor() {
        super(...arguments);
        this.cdsMotionChange = new EventEmitter();
    }
    get cdsMotion() {
        return this.element['cdsMotion'];
    }
    set cdsMotion(value) {
        this.element['cdsMotion'] = value;
    }
    get i18n() {
        return this.element['i18n'];
    }
    set i18n(value) {
        this.element['i18n'] = value;
    }
    get expanded() {
        return this.element['expanded'];
    }
    set expanded(value) {
        if (value === '') {
            this.element['expanded'] = true;
            return;
        }
        this.element['expanded'] = value;
    }
    get active() {
        return this.element['active'];
    }
    set active(value) {
        if (value === '') {
            this.element['active'] = true;
            return;
        }
        this.element['active'] = value;
    }
    get hasFocus() {
        return this.element['hasFocus'];
    }
    set hasFocus(value) {
        if (value === '') {
            this.element['hasFocus'] = true;
            return;
        }
        this.element['hasFocus'] = value;
    }
}
CdsNavigationGroupDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-navigation-group' },] }
];
CdsNavigationGroupDirective.propDecorators = {
    cdsMotion: [{ type: Input }],
    i18n: [{ type: Input }],
    expanded: [{ type: Input }],
    active: [{ type: Input }],
    hasFocus: [{ type: Input }],
    cdsMotionChange: [{ type: Output }]
};
