// @cds/angular/esm2015/cds/components/navigation/cds-navigation-item.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsNavigationItemDirective extends BaseCdsDirective {
    get i18n() {
        return this.element['i18n'];
    }
    set i18n(value) {
        this.element['i18n'] = value;
    }
    get active() {
        return this.element['active'];
    }
    set active(value) {
        if (value === '') {
            this.element['active'] = true;
            return;
        }
        this.element['active'] = value;
    }
    get disabled() {
        return this.element['disabled'];
    }
    set disabled(value) {
        if (value === '') {
            this.element['disabled'] = true;
            return;
        }
        this.element['disabled'] = value;
    }
    get role() {
        return this.element['role'];
    }
    set role(value) {
        this.element['role'] = value;
    }
    get expanded() {
        return this.element['expanded'];
    }
    set expanded(value) {
        if (value === '') {
            this.element['expanded'] = true;
            return;
        }
        this.element['expanded'] = value;
    }
    get groupItem() {
        return this.element['groupItem'];
    }
    set groupItem(value) {
        if (value === '') {
            this.element['groupItem'] = true;
            return;
        }
        this.element['groupItem'] = value;
    }
    get hasFocus() {
        return this.element['hasFocus'];
    }
    set hasFocus(value) {
        this.element['hasFocus'] = value;
    }
    get focusElement() {
        return this.element['focusElement'];
    }
    set focusElement(value) {
        this.element['focusElement'] = value;
    }
    get itemText() {
        return this.element['itemText'];
    }
    set itemText(value) {
        this.element['itemText'] = value;
    }
}
CdsNavigationItemDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-navigation-item' },] }
];
CdsNavigationItemDirective.propDecorators = {
    i18n: [{ type: Input }],
    active: [{ type: Input }],
    disabled: [{ type: Input }],
    role: [{ type: Input }],
    expanded: [{ type: Input }],
    groupItem: [{ type: Input }],
    hasFocus: [{ type: Input }],
    focusElement: [{ type: Input }],
    itemText: [{ type: Input }]
};
