// ngx-chart/__ivy_ngcc__/fesm2015/ngx-chart.js
import { __decorate } from 'tslib';
import { EventEmitter, Input, ViewChildren, Output, Component, ElementRef, Renderer2, HostListener, Directive, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

const _c0 = ["el"];
function PieChartComponent__svg_path_2_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "path", 4, 5);
    ɵngcc0.ɵɵlistener("click", function PieChartComponent__svg_path_2_Template__svg_path_click_0_listener() { const restoredCtx = ɵngcc0.ɵɵrestoreView(_r6); const item_r2 = restoredCtx.$implicit; const ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.getItemClicked(item_r2); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r2 = ctx.$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("tooltip", item_r2.name);
} }
const _c1 = function (a0) { return { "background": a0 }; };
function PieChartComponent_div_3_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 8);
    ɵngcc0.ɵɵelement(1, "div", 9);
    ɵngcc0.ɵɵelementStart(2, "span", 10);
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "span", 11);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r8 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c1, item_r8.color));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r8.name);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r8.value);
} }
function PieChartComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵnamespaceHTML();
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵelementStart(1, "h4");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, PieChartComponent_div_3_div_3_Template, 6, 5, "div", 7);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.chartOptions.legendTitle + " : " + ctx_r1.totalSum);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.legendData);
} }
const _c2 = function (a0, a1) { return { "height": a0, "width": a1 }; };
function BarChartComponent__svg_line_4_Template(rf, ctx) { if (rf & 1) {
    const _r4 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "line", 5);
    ɵngcc0.ɵɵlistener("click", function BarChartComponent__svg_line_4_Template__svg_line_click_0_listener() { const restoredCtx = ɵngcc0.ɵɵrestoreView(_r4); const line_r2 = restoredCtx.$implicit; const ctx_r3 = ɵngcc0.ɵɵnextContext(); return ctx_r3.getItemClicked(line_r2); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const line_r2 = ctx.$implicit;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("tooltip", line_r2.name);
    ɵngcc0.ɵɵattribute("x1", line_r2.x1)("y1", line_r2.y1)("x2", line_r2.x2)("y2", line_r2.y2)("stroke", line_r2.color)("stroke-linecap", ctx_r0.barLineType)("title", line_r2.name);
} }
function BarChartComponent_div_5_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 8);
    ɵngcc0.ɵɵelement(1, "div", 9);
    ɵngcc0.ɵɵelementStart(2, "span", 10);
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "span", 11);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r6 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c1, item_r6.color));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r6.name);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r6.value);
} }
function BarChartComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵnamespaceHTML();
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵelementStart(1, "h4");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, BarChartComponent_div_5_div_3_Template, 6, 5, "div", 7);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.chartOptions.legendTitle + " : " + ctx_r1.totalSum);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.chartData);
} }
function DonutChartComponent__svg_path_2_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵelementStart(0, "path", 4, 5);
    ɵngcc0.ɵɵlistener("click", function DonutChartComponent__svg_path_2_Template__svg_path_click_0_listener() { const restoredCtx = ɵngcc0.ɵɵrestoreView(_r6); const item_r2 = restoredCtx.$implicit; const ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.getItemClicked(item_r2); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r2 = ctx.$implicit;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("tooltip", item_r2.name);
    ɵngcc0.ɵɵattribute("stroke-width", ctx_r0.view.donutSize);
} }
function DonutChartComponent_div_3_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 8);
    ɵngcc0.ɵɵelement(1, "div", 9);
    ɵngcc0.ɵɵelementStart(2, "span", 10);
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "span", 11);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r8 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c1, item_r8.color));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r8.name);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r8.value);
} }
function DonutChartComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵnamespaceSVG();
    ɵngcc0.ɵɵnamespaceHTML();
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵelementStart(1, "h4");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, DonutChartComponent_div_3_div_3_Template, 6, 5, "div", 7);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.chartOptions.legendTitle + " : " + ctx_r1.totalSum);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.legendData);
} }
let PieChartComponent = class PieChartComponent {
    constructor() {
        this.selectedItem = new EventEmitter();
        this.chartOptions = {
            showLegend: true,
            legendTitle: 'Total',
        };
        this.view = {
            height: 200,
            width: 200,
            radius: 80
        };
        this.chartView = [];
        this.totalSum = 0;
        this.processedData = [];
        this.legendData = [];
    }
    ngOnInit() {
        this.chartView.push(this.view.height + 'px');
        this.chartView.push(this.view.width + 'px');
        if (this.chartData) {
            // Get total number of records
            this.totalSum = this.chartData.reduce(function (a, b) {
                return a + b.value;
            }, 0);
            //generate Data for piechart
            let prevAngle = 0;
            this.chartData.map((x, index) => {
                let legend = {
                    name: x.name,
                    value: x.value,
                    color: x.color
                };
                this.legendData.push(legend);
                let percentage = this.getPercentage(x.value, this.totalSum);
                if (percentage > 0) {
                    let circlePercentage = percentage / 10 * 36;
                    let pieData = {
                        color: x.color,
                        a1: prevAngle,
                        a2: prevAngle + circlePercentage,
                        name: x.name
                    };
                    prevAngle = prevAngle + circlePercentage;
                    this.processedData.push(pieData);
                }
            });
        }
    }
    ngAfterViewInit() {
        // create svg 
        if (this.processedData.length > 0) {
            this.span.map((item, index) => {
                this.processedData[index].a2 = this.processedData[index].a2 >= 360 ? 359.9 : this.processedData[index].a2;
                item.nativeElement.setAttribute('d', this.describeArc(this.view.height / 2, this.view.height / 2, this.view.radius, this.processedData[index].a1, this.processedData[index].a2));
                item.nativeElement.setAttribute('fill', this.processedData[index].color);
            });
        }
    }
    polarToCartesian(centerX, centerY, radius, angleInDegrees) {
        let angleInRadians = (angleInDegrees - 90) * Math.PI / 180.0;
        return {
            x: centerX + (radius * Math.cos(angleInRadians)),
            y: centerY + (radius * Math.sin(angleInRadians))
        };
    }
    describeArc(x, y, radius, startAngle, endAngle) {
        let start = this.polarToCartesian(x, y, radius, endAngle);
        let end = this.polarToCartesian(x, y, radius, startAngle);
        let arcSweep = endAngle - startAngle <= 180 ? "0" : "1";
        let d = [
            "M", start.x, start.y,
            "A", radius, radius, 0, arcSweep, 0, end.x, end.y,
            "L", x, y,
            "L", start.x, start.y
        ].join(" ");
        return d;
    }
    getPercentage(partialValue, totalValue) {
        return (100 * partialValue) / totalValue;
    }
    getItemClicked(item) {
        let selected = {
            name: item.name,
            value: item.value
        };
        this.selectedItem.emit(selected);
    }
};
PieChartComponent.ɵfac = function PieChartComponent_Factory(t) { return new (t || PieChartComponent)(); };
PieChartComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: PieChartComponent, selectors: [["ngx-pie-chart"]], viewQuery: function PieChartComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 5);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.span = _t);
    } }, inputs: { chartOptions: "chartOptions", view: "view", chartData: "chartData" }, outputs: { selectedItem: "selectedItem" }, decls: 4, vars: 6, consts: [[1, "chart-view", "ngx-pie"], [1, "chart-wrapper", 3, "ngStyle"], ["class", "pie-chart-path", "chartTooltip", "", 3, "tooltip", "click", 4, "ngFor", "ngForOf"], ["class", "legend", 4, "ngIf"], ["chartTooltip", "", 1, "pie-chart-path", 3, "tooltip", "click"], ["el", ""], [1, "legend"], ["class", "legend-item", 4, "ngFor", "ngForOf"], [1, "legend-item"], [1, "legend-item-colour", 3, "ngStyle"], [1, "legend-item-name"], [1, "legend-item-value"]], template: function PieChartComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelementStart(1, "svg", 1);
        ɵngcc0.ɵɵtemplate(2, PieChartComponent__svg_path_2_Template, 2, 1, "path", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(3, PieChartComponent_div_3_Template, 4, 2, "div", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction2(3, _c2, ctx.chartView[0], ctx.chartView[1]));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.processedData);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.chartOptions.showLegend);
    } }, directives: function () { return [ɵngcc1.NgStyle, ɵngcc1.NgForOf, ɵngcc1.NgIf, TooltipDirective]; }, styles: ["path.pie-chart-path[_ngcontent-%COMP%]:hover{cursor:pointer}"] });
__decorate([
    Input()
], PieChartComponent.prototype, "chartData", void 0);
__decorate([
    ViewChildren('el')
], PieChartComponent.prototype, "span", void 0);
__decorate([
    Output()
], PieChartComponent.prototype, "selectedItem", void 0);
__decorate([
    Input()
], PieChartComponent.prototype, "chartOptions", void 0);
__decorate([
    Input()
], PieChartComponent.prototype, "view", void 0);

let BarChartComponent = class BarChartComponent {
    constructor() {
        this.selectedItem = new EventEmitter();
        this.chartData = [];
        this.chartOptions = {
            roundedCorners: false,
            isHorizontal: false,
            showLegend: true,
            legendTitle: 'Total',
        };
        this.view = {
            height: 200,
            width: 200
        };
        this.lines = [];
        this.totalSum = 0;
        this.isHorizontal = true;
        this.barLineType = 'round';
        this.chartView = [];
        this.scale = 100;
    }
    ngOnInit() {
        this.chartView.push(this.view.height + 'px');
        this.chartView.push(this.view.width + 'px');
        this.barLineType = this.chartOptions['roundedCorners'] ? 'round' : 'butt';
        this.isHorizontal = this.chartOptions['isHorizontal'] ? true : false;
        this.scale = this.isHorizontal ? this.view.width : this.view.height;
        this.getTotalSum(this.chartData);
        let value = Math.max.apply(Math, this.chartData.map(function (o) {
            return o.value;
        }));
        if (value > this.scale) {
            let normalizedValue = value / this.scale;
            this.chartData.map((y, index) => {
                this.chartData[index]['normalized'] = Number(y.value) / normalizedValue;
            });
        }
        else {
            let normalizedValue = this.scale / value;
            this.chartData.map((y, index) => {
                this.chartData[index]['normalized'] = Number(y.value) * normalizedValue;
            });
        }
        this.lines = this.isHorizontal ? this.calculateHorizontalBarLines(this.chartData) : this.calculateVerticalBarLines(this.chartData);
    }
    ngAfterViewInit() { }
    getItemClicked(line) {
        let selectedbar = {
            name: line.name,
            value: line.value
        };
        this.selectedItem.emit(JSON.stringify(selectedbar));
    }
    calculateHorizontalBarLines(graphData) {
        let barLines = [];
        graphData.map((x, index) => {
            let background = {
                x1: 10,
                y1: index * 20 + 20,
                y2: index * 20 + 20,
                x2: this.view.width,
                color: '#EBEBEB',
                name: x.name,
                value: x.value
            };
            barLines.push(background);
            if (x.value > 0) {
                let line = {
                    x1: 10,
                    y1: index * 20 + 20,
                    y2: index * 20 + 20,
                    x2: x.normalized < 10 ? this.barLineType == 'butt' ? 11 : 10 : x.normalized,
                    color: x.color,
                    name: x.name,
                    value: x.value
                };
                barLines.push(line);
            }
        });
        return barLines;
    }
    calculateVerticalBarLines(graphData) {
        let barLines = [];
        graphData.map((x, index) => {
            let background = {
                x1: index * 20 + 20,
                y1: 10,
                x2: index * 20 + 20,
                y2: this.scale,
                color: '#EBEBEB',
                name: x.name,
                value: x.value
            };
            barLines.push(background);
            if (x.value > 0) {
                let line = {
                    x1: index * 20 + 20,
                    y1: (this.scale + 10) - x.normalized > this.scale ? this.barLineType == 'butt' ? (this.scale - 1) : this.scale : (this.scale + 10) - x.normalized,
                    x2: index * 20 + 20,
                    y2: this.scale,
                    color: x.color,
                    name: x.name,
                    value: x.value
                };
                barLines.push(line);
            }
        });
        console.log(barLines);
        return barLines;
    }
    getTotalSum(chartData) {
        // Get total number of records
        this.totalSum = chartData.reduce(function (a, b) {
            return a + parseInt(b.value);
        }, 0);
    }
};
BarChartComponent.ɵfac = function BarChartComponent_Factory(t) { return new (t || BarChartComponent)(); };
BarChartComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: BarChartComponent, selectors: [["ngx-bar-chart"]], inputs: { chartData: "chartData", chartOptions: "chartOptions", view: "view" }, outputs: { selectedItem: "selectedItem" }, decls: 6, vars: 7, consts: [[1, "chart-view", "ngx-bar"], [1, "chart-wrapper", 3, "ngStyle"], ["id", "lines"], ["stroke-width", "10", "class", "line", "chartTooltip", "", 3, "tooltip", "click", 4, "ngFor", "ngForOf"], ["class", "legend", 4, "ngIf"], ["stroke-width", "10", "chartTooltip", "", 1, "line", 3, "tooltip", "click"], [1, "legend"], ["class", "legend-item", 4, "ngFor", "ngForOf"], [1, "legend-item"], [1, "legend-item-colour", 3, "ngStyle"], [1, "legend-item-name"], [1, "legend-item-value"]], template: function BarChartComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelementStart(2, "svg", 2);
        ɵngcc0.ɵɵelementStart(3, "g");
        ɵngcc0.ɵɵtemplate(4, BarChartComponent__svg_line_4_Template, 1, 8, "line", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(5, BarChartComponent_div_5_Template, 4, 2, "div", 4);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction2(4, _c2, ctx.chartView[0], ctx.chartView[1]));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("viewBox", "0 0 " + (ctx.isHorizontal ? ctx.scale + 40 : ctx.lines[ctx.lines.length - 1].x1 + 20) + " " + (ctx.isHorizontal ? ctx.lines[ctx.lines.length - 1].y2 + 20 : ctx.scale + 20));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.lines);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.chartOptions.showLegend);
    } }, directives: function () { return [ɵngcc1.NgStyle, ɵngcc1.NgForOf, ɵngcc1.NgIf, TooltipDirective]; }, styles: [".line[_ngcontent-%COMP%]{cursor:pointer}"] });
__decorate([
    Output()
], BarChartComponent.prototype, "selectedItem", void 0);
__decorate([
    Input()
], BarChartComponent.prototype, "chartData", void 0);
__decorate([
    Input()
], BarChartComponent.prototype, "chartOptions", void 0);
__decorate([
    Input()
], BarChartComponent.prototype, "view", void 0);

let DonutChartComponent = class DonutChartComponent {
    constructor() {
        this.selectedItem = new EventEmitter();
        this.chartOptions = {
            showLegend: true,
            legendTitle: 'Total',
        };
        this.view = {
            height: 200,
            width: 200,
            radius: 80,
            donutSize: 20
        };
        this.totalSum = 0;
        this.processedData = [];
        this.legendData = [];
        this.chartView = [];
    }
    ngOnInit() {
        if (this.chartData) {
            this.view['donutSize'] = this.view['donutSize'] ? this.view['donutSize'] : 20;
            this.chartView.push(this.view.height + 'px');
            this.chartView.push(this.view.width + 'px');
            // Get total number of records
            this.totalSum = this.chartData.reduce(function (a, b) {
                return a + b.value;
            }, 0);
            //generate Data for list
            let prevAngle = 0;
            this.chartData.map((x, index) => {
                let legend = {
                    name: x.name,
                    value: x.value,
                    color: x.color
                };
                this.legendData.push(legend);
                let percentage = this.getPercentage(x.value, this.totalSum);
                if (percentage > 0) {
                    let circlePercentage = percentage / 10 * 36;
                    let pieData = {
                        color: x.color,
                        a1: prevAngle,
                        a2: prevAngle + circlePercentage,
                        name: x.name
                    };
                    prevAngle = prevAngle + circlePercentage;
                    this.processedData.push(pieData);
                }
            });
        }
    }
    getPercentage(partialValue, totalValue) {
        return (100 * partialValue) / totalValue;
    }
    ngAfterViewInit() {
        //create svg
        if (this.processedData.length > 0) {
            this.span.map((item, index) => {
                item.nativeElement.setAttribute('d', this.describeArc(this.view.height / 2, this.view.width / 2, this.view.radius, this.processedData[index].a1, this.processedData[index].a2 == 360 ? 359.99 : this.processedData[index].a2));
                item.nativeElement.setAttribute('stroke', this.processedData[index].color);
            });
        }
    }
    polarToCartesian(centerX, centerY, radius, angleInDegrees) {
        let angleInRadians = (angleInDegrees - 90) * Math.PI / 180.0;
        return {
            x: centerX + (radius * Math.cos(angleInRadians)),
            y: centerY + (radius * Math.sin(angleInRadians))
        };
    }
    describeArc(x, y, radius, startAngle, endAngle) {
        let start = this.polarToCartesian(x, y, radius, endAngle);
        let end = this.polarToCartesian(x, y, radius, startAngle);
        let largeArcFlag = endAngle - startAngle <= 180 ? "0" : "1";
        let d = [
            "M", start.x, start.y,
            "A", radius, radius, 0, largeArcFlag, 0, end.x, end.y
        ].join(" ");
        return d;
    }
    getItemClicked(item) {
        let selected = {
            name: item.name,
            value: item.value
        };
        this.selectedItem.emit(selected);
    }
    show(status) {
    }
};
DonutChartComponent.ɵfac = function DonutChartComponent_Factory(t) { return new (t || DonutChartComponent)(); };
DonutChartComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: DonutChartComponent, selectors: [["ngx-donut-chart"]], viewQuery: function DonutChartComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 5);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.span = _t);
    } }, inputs: { chartOptions: "chartOptions", view: "view", chartData: "chartData" }, outputs: { selectedItem: "selectedItem" }, decls: 4, vars: 6, consts: [[1, "chart-view", "ngx-donut"], [1, "chart-wrapper", 3, "ngStyle"], ["width", "100%", "height", "100%", "fill", "none", "chartTooltip", "", 3, "tooltip", "click", 4, "ngFor", "ngForOf"], ["class", "legend", 4, "ngIf"], ["width", "100%", "height", "100%", "fill", "none", "chartTooltip", "", 3, "tooltip", "click"], ["el", ""], [1, "legend"], ["class", "legend-item", 4, "ngFor", "ngForOf"], [1, "legend-item"], [1, "legend-item-colour", 3, "ngStyle"], [1, "legend-item-name"], [1, "legend-item-value"]], template: function DonutChartComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵnamespaceSVG();
        ɵngcc0.ɵɵelementStart(1, "svg", 1);
        ɵngcc0.ɵɵtemplate(2, DonutChartComponent__svg_path_2_Template, 2, 2, "path", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(3, DonutChartComponent_div_3_Template, 4, 2, "div", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction2(3, _c2, ctx.chartView[0], ctx.chartView[1]));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.processedData);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.chartOptions.showLegend);
    } }, directives: function () { return [ɵngcc1.NgStyle, ɵngcc1.NgForOf, ɵngcc1.NgIf, TooltipDirective]; }, styles: ["path[_ngcontent-%COMP%]:hover{cursor:pointer}"] });
__decorate([
    Input()
], DonutChartComponent.prototype, "chartData", void 0);
__decorate([
    Output()
], DonutChartComponent.prototype, "selectedItem", void 0);
__decorate([
    ViewChildren('el')
], DonutChartComponent.prototype, "span", void 0);
__decorate([
    Input()
], DonutChartComponent.prototype, "chartOptions", void 0);
__decorate([
    Input()
], DonutChartComponent.prototype, "view", void 0);

let TooltipDirective = class TooltipDirective {
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.placement = 'top';
        this.offset = 10;
    }
    onMouseEnter() {
        if (!this.tooltip) {
            this.show();
        }
    }
    onMouseLeave() {
        if (this.tooltip) {
            this.hide();
        }
    }
    show() {
        this.create();
        this.setPosition();
        this.renderer.addClass(this.tooltip, 'ngx-chart-tooltip-show');
    }
    hide() {
        this.renderer.removeClass(this.tooltip, 'ngx-chart-tooltip-show');
        this.renderer.removeChild(document.body, this.tooltip);
        this.tooltip = null;
    }
    create() {
        this.tooltip = this.renderer.createElement('span');
        this.renderer.appendChild(this.tooltip, this.renderer.createText(this.tooltipTitle) // textNode
        );
        this.renderer.appendChild(document.body, this.tooltip);
        this.renderer.addClass(this.tooltip, 'ngx-chart-tooltip');
        this.renderer.addClass(this.tooltip, `ngx-chart-tooltip-${this.placement}`);
    }
    setPosition() {
        const hostPos = this.el.nativeElement.getBoundingClientRect();
        const tooltipPos = this.tooltip.getBoundingClientRect();
        const scrollPos = window.pageYOffset || document.documentElement.scrollTop || document.body.scrollTop || 0;
        let top, left;
        if (this.placement === 'bottom') {
            top = hostPos.bottom + this.offset;
            left = hostPos.left + (hostPos.width - tooltipPos.width) / 2;
        }
        else if (this.placement === 'left') {
            top = hostPos.top + (hostPos.height - tooltipPos.height) / 2;
            left = hostPos.left - tooltipPos.width - this.offset;
        }
        else if (this.placement === 'right') {
            top = hostPos.top + (hostPos.height - tooltipPos.height) / 2;
            left = hostPos.right + this.offset;
        }
        else {
            // top
            top = hostPos.top - tooltipPos.height - this.offset;
            left = hostPos.left + (hostPos.width - tooltipPos.width) / 2;
        }
        this.renderer.setStyle(this.tooltip, 'top', `${top + scrollPos}px`);
        this.renderer.setStyle(this.tooltip, 'left', `${left}px`);
    }
};
TooltipDirective.ɵfac = function TooltipDirective_Factory(t) { return new (t || TooltipDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2)); };
TooltipDirective.ɵdir = /*@__PURE__*/ ɵngcc0.ɵɵdefineDirective({ type: TooltipDirective, selectors: [["", "chartTooltip", ""]], hostBindings: function TooltipDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("mouseenter", function TooltipDirective_mouseenter_HostBindingHandler() { return ctx.onMouseEnter(); })("mouseleave", function TooltipDirective_mouseleave_HostBindingHandler() { return ctx.onMouseLeave(); });
    } }, inputs: { placement: "placement", tooltipTitle: ["tooltip", "tooltipTitle"] } });
TooltipDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
__decorate([
    Input('tooltip')
], TooltipDirective.prototype, "tooltipTitle", void 0);
__decorate([
    Input()
], TooltipDirective.prototype, "placement", void 0);
__decorate([
    HostListener('mouseenter')
], TooltipDirective.prototype, "onMouseEnter", null);
__decorate([
    HostListener('mouseleave')
], TooltipDirective.prototype, "onMouseLeave", null);

let NgxChartModule = class NgxChartModule {
};
NgxChartModule.ɵfac = function NgxChartModule_Factory(t) { return new (t || NgxChartModule)(); };
NgxChartModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgxChartModule });
NgxChartModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(PieChartComponent, [{
        type: Component,
        args: [{
                selector: 'ngx-pie-chart',
                template: "<div class=\"chart-view ngx-pie\">\n  <svg class=\"chart-wrapper\" [ngStyle]=\"{'height': chartView[0],'width': chartView[1]}\">\n    <path #el class=\"pie-chart-path\" chartTooltip tooltip={{item.name}} (click)=\"getItemClicked(item)\" *ngFor=\"let item of processedData; let i=index\" />\n  </svg>\n  <div class=\"legend\" *ngIf=\"chartOptions.showLegend\">\n    <h4>{{chartOptions.legendTitle +' : '+  totalSum}}</h4>\n    <div class=\"legend-item\" *ngFor=\"let item of legendData; let i=index\">\n      <div class=\"legend-item-colour\" [ngStyle]=\"{'background': item.color}\"></div><span class=\"legend-item-name\" >{{item.name}}</span><span class=\"legend-item-value\" >{{item.value}}</span>\n    </div>\n  </div>\n</div>",
                styles: ["path.pie-chart-path:hover{cursor:pointer}"]
            }]
    }], function () { return []; }, { selectedItem: [{
            type: Output
        }], chartOptions: [{
            type: Input
        }], view: [{
            type: Input
        }], chartData: [{
            type: Input
        }], span: [{
            type: ViewChildren,
            args: ['el']
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BarChartComponent, [{
        type: Component,
        args: [{
                selector: 'ngx-bar-chart',
                template: "<div class=\"chart-view  ngx-bar\">\r\n  <div class=\"chart-wrapper\" [ngStyle]=\"{'height': chartView[0],'width': chartView[1]}\">\r\n    <svg id=\"lines\" [attr.viewBox]=\"'0 0 '+ (isHorizontal ? scale+40 : (lines[lines.length-1].x1 + 20))+' ' + (isHorizontal ? (lines[lines.length-1].y2 + 20) : (scale+20))\">\r\n      <g>\r\n        <line *ngFor=\"let line of lines\" stroke-width=\"10\" class=\"line\" [attr.x1]=\"line.x1\" [attr.y1]=\"line.y1 \"\r\n          [attr.x2]=\"line.x2\" [attr.y2]=\"line.y2\" [attr.stroke]=\"line.color\" \u00A0\u00A0\u00A0\u00A0\u00A0\u00A0\u00A0 [attr.stroke-linecap]=\"barLineType\"\r\n          [attr.title]=\"line.name\" (click)=\"getItemClicked(line)\" chartTooltip tooltip={{line.name}} />\r\n      </g>\r\n    </svg>\r\n  </div>\r\n  <div class=\"legend\" *ngIf=\"chartOptions.showLegend\">\r\n    <h4>{{chartOptions.legendTitle +' : '+ totalSum}}</h4>\r\n    <div class=\"legend-item\" *ngFor=\"let item of chartData; let i=index\">\r\n      <div class=\"legend-item-colour\" [ngStyle]=\"{'background': item.color}\"></div><span\r\n        class=\"legend-item-name\">{{item.name}}</span><span class=\"legend-item-value\">{{item.value}}</span>\r\n    </div>\r\n  </div>\r\n\r\n</div>",
                styles: [".line{cursor:pointer}"]
            }]
    }], function () { return []; }, { selectedItem: [{
            type: Output
        }], chartData: [{
            type: Input
        }], chartOptions: [{
            type: Input
        }], view: [{
            type: Input
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DonutChartComponent, [{
        type: Component,
        args: [{
                selector: 'ngx-donut-chart',
                template: "<div class=\"chart-view ngx-donut\">\n  <svg class=\"chart-wrapper\" [ngStyle]=\"{'height': chartView[0],'width': chartView[1]}\">\n    <path width=\"100%\" height=\"100%\" #el fill=\"none\" [attr.stroke-width]=\"this.view.donutSize\" *ngFor=\"let item of processedData; let i=index\"\n      (click)=\"getItemClicked(item)\" chartTooltip tooltip={{item.name}}/>\n\n  </svg>\n  <div class=\"legend\" *ngIf=\"chartOptions.showLegend\">\n    <h4>{{chartOptions.legendTitle +' : '+  totalSum}}</h4>\n    <div class=\"legend-item\" *ngFor=\"let item of legendData; let i=index\" >\n      <div class=\"legend-item-colour\" [ngStyle]=\"{'background': item.color}\"></div><span class=\"legend-item-name\" >{{item.name}}</span><span class=\"legend-item-value\" >{{item.value}}</span>\n    </div>\n  </div>\n</div>",
                styles: ["path:hover{cursor:pointer}"]
            }]
    }], function () { return []; }, { selectedItem: [{
            type: Output
        }], chartOptions: [{
            type: Input
        }], view: [{
            type: Input
        }], chartData: [{
            type: Input
        }], span: [{
            type: ViewChildren,
            args: ['el']
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TooltipDirective, [{
        type: Directive,
        args: [{
                selector: '[chartTooltip]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }]; }, { placement: [{
            type: Input
        }], onMouseEnter: [{
            type: HostListener,
            args: ['mouseenter']
        }], onMouseLeave: [{
            type: HostListener,
            args: ['mouseleave']
        }], tooltipTitle: [{
            type: Input,
            args: ['tooltip']
        }] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxChartModule, [{
        type: NgModule,
        args: [{
                declarations: [PieChartComponent, BarChartComponent, DonutChartComponent, TooltipDirective],
                imports: [
                    CommonModule
                ],
                exports: [PieChartComponent, BarChartComponent, DonutChartComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxChartModule, { declarations: function () { return [PieChartComponent, BarChartComponent, DonutChartComponent, TooltipDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [PieChartComponent, BarChartComponent, DonutChartComponent]; } }); })();

/*
 * Public API Surface of ngx-chart
 */

/**
 * Generated bundle index. Do not edit.
 */

export { BarChartComponent, DonutChartComponent, NgxChartModule, PieChartComponent, TooltipDirective as ɵa };
