// @cds/angular/esm2015/cds/components/pagination/cds-pagination-button.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsPaginationButtonDirective extends BaseCdsDirective {
    get action() {
        return this.element['action'];
    }
    set action(value) {
        this.element['action'] = value;
    }
    get cdsIcon() {
        return this.element['cdsIcon'];
    }
    set cdsIcon(value) {
        this.element['cdsIcon'] = value;
    }
    get readonly() {
        return this.element['readonly'];
    }
    set readonly(value) {
        if (value === '') {
            this.element['readonly'] = true;
            return;
        }
        this.element['readonly'] = value;
    }
    get type() {
        return this.element['type'];
    }
    set type(value) {
        this.element['type'] = value;
    }
    get name() {
        return this.element['name'];
    }
    set name(value) {
        this.element['name'] = value;
    }
    get value() {
        return this.element['value'];
    }
    set value(value) {
        this.element['value'] = value;
    }
    get disabled() {
        return this.element['disabled'];
    }
    set disabled(value) {
        if (value === '') {
            this.element['disabled'] = true;
            return;
        }
        this.element['disabled'] = value;
    }
    get ariaDisabled() {
        return this.element['ariaDisabled'];
    }
    set ariaDisabled(value) {
        this.element['ariaDisabled'] = value;
    }
}
CdsPaginationButtonDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-pagination-button' },] }
];
CdsPaginationButtonDirective.propDecorators = {
    action: [{ type: Input }],
    cdsIcon: [{ type: Input }],
    readonly: [{ type: Input }],
    type: [{ type: Input }],
    name: [{ type: Input }],
    value: [{ type: Input }],
    disabled: [{ type: Input }],
    ariaDisabled: [{ type: Input }]
};
