// @cds/angular/esm2015/cds/components/modal/cds-modal.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input, Output, EventEmitter } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsModalDirective extends BaseCdsDirective {
    constructor() {
        super(...arguments);
        this.cdsMotionChange = new EventEmitter();
    }
    get i18n() {
        return this.element['i18n'];
    }
    set i18n(value) {
        this.element['i18n'] = value;
    }
    get closable() {
        return this.element['closable'];
    }
    set closable(value) {
        if (value === '') {
            this.element['closable'] = true;
            return;
        }
        this.element['closable'] = value;
    }
    get size() {
        return this.element['size'];
    }
    set size(value) {
        this.element['size'] = value;
    }
    get cdsMotion() {
        return this.element['cdsMotion'];
    }
    set cdsMotion(value) {
        this.element['cdsMotion'] = value;
    }
    get ariaModal() {
        return this.element['ariaModal'];
    }
    set ariaModal(value) {
        this.element['ariaModal'] = value;
    }
    get role() {
        return this.element['role'];
    }
    set role(value) {
        this.element['role'] = value;
    }
    get shadowRootOptions() {
        return this.element['shadowRootOptions'];
    }
    set shadowRootOptions(value) {
        this.element['shadowRootOptions'] = value;
    }
    get focusTrap() {
        return this.element['focusTrap'];
    }
    set focusTrap(value) {
        this.element['focusTrap'] = value;
    }
    get topReboundElement() {
        return this.element['topReboundElement'];
    }
    set topReboundElement(value) {
        this.element['topReboundElement'] = value;
    }
    get bottomReboundElement() {
        return this.element['bottomReboundElement'];
    }
    set bottomReboundElement(value) {
        this.element['bottomReboundElement'] = value;
    }
    get hidden() {
        return this.element['hidden'];
    }
    set hidden(value) {
        if (value === '') {
            this.element['hidden'] = true;
            return;
        }
        this.element['hidden'] = value;
    }
    get focusTrapId() {
        return this.element['focusTrapId'];
    }
    set focusTrapId(value) {
        this.element['focusTrapId'] = value;
    }
}
CdsModalDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-modal' },] }
];
CdsModalDirective.propDecorators = {
    i18n: [{ type: Input }],
    closable: [{ type: Input }],
    size: [{ type: Input }],
    cdsMotion: [{ type: Input }],
    ariaModal: [{ type: Input }],
    role: [{ type: Input }],
    shadowRootOptions: [{ type: Input }],
    focusTrap: [{ type: Input }],
    topReboundElement: [{ type: Input }],
    bottomReboundElement: [{ type: Input }],
    hidden: [{ type: Input }],
    focusTrapId: [{ type: Input }],
    cdsMotionChange: [{ type: Output }]
};
