// @cds/angular/esm2015/cds/components/button/cds-button.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsButtonDirective extends BaseCdsDirective {
    get action() {
        return this.element['action'];
    }
    set action(value) {
        this.element['action'] = value;
    }
    get status() {
        return this.element['status'];
    }
    set status(value) {
        this.element['status'] = value;
    }
    get size() {
        return this.element['size'];
    }
    set size(value) {
        this.element['size'] = value;
    }
    get block() {
        return this.element['block'];
    }
    set block(value) {
        if (value === '') {
            this.element['block'] = true;
            return;
        }
        this.element['block'] = value;
    }
    get loadingState() {
        return this.element['loadingState'];
    }
    set loadingState(value) {
        this.element['loadingState'] = value;
    }
    get readonly() {
        return this.element['readonly'];
    }
    set readonly(value) {
        if (value === '') {
            this.element['readonly'] = true;
            return;
        }
        this.element['readonly'] = value;
    }
    get type() {
        return this.element['type'];
    }
    set type(value) {
        this.element['type'] = value;
    }
    get name() {
        return this.element['name'];
    }
    set name(value) {
        this.element['name'] = value;
    }
    get value() {
        return this.element['value'];
    }
    set value(value) {
        this.element['value'] = value;
    }
    get disabled() {
        return this.element['disabled'];
    }
    set disabled(value) {
        if (value === '') {
            this.element['disabled'] = true;
            return;
        }
        this.element['disabled'] = value;
    }
    get ariaDisabled() {
        return this.element['ariaDisabled'];
    }
    set ariaDisabled(value) {
        this.element['ariaDisabled'] = value;
    }
}
CdsButtonDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-button' },] }
];
CdsButtonDirective.propDecorators = {
    action: [{ type: Input }],
    status: [{ type: Input }],
    size: [{ type: Input }],
    block: [{ type: Input }],
    loadingState: [{ type: Input }],
    readonly: [{ type: Input }],
    type: [{ type: Input }],
    name: [{ type: Input }],
    value: [{ type: Input }],
    disabled: [{ type: Input }],
    ariaDisabled: [{ type: Input }]
};
