// @cds/angular/esm2015/cds/components/forms/cds-internal-control-group.directive.js
/*
 * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import { BaseCdsDirective } from '../../cds-base';
export class CdsInternalControlGroupDirective extends BaseCdsDirective {
    get status() {
        return this.element['status'];
    }
    set status(value) {
        this.element['status'] = value;
    }
    get layout() {
        return this.element['layout'];
    }
    set layout(value) {
        this.element['layout'] = value;
    }
    get controlAlign() {
        return this.element['controlAlign'];
    }
    set controlAlign(value) {
        this.element['controlAlign'] = value;
    }
    get disabled() {
        return this.element['disabled'];
    }
    set disabled(value) {
        if (value === '') {
            this.element['disabled'] = true;
            return;
        }
        this.element['disabled'] = value;
    }
    get controlWidth() {
        return this.element['controlWidth'];
    }
    set controlWidth(value) {
        this.element['controlWidth'] = value;
    }
    get responsive() {
        return this.element['responsive'];
    }
    set responsive(value) {
        if (value === '') {
            this.element['responsive'] = true;
            return;
        }
        this.element['responsive'] = value;
    }
    get layoutStable() {
        return this.element['layoutStable'];
    }
    set layoutStable(value) {
        if (value === '') {
            this.element['layoutStable'] = true;
            return;
        }
        this.element['layoutStable'] = value;
    }
}
CdsInternalControlGroupDirective.decorators = [
    { type: Directive, args: [{ selector: 'cds-internal-control-group' },] }
];
CdsInternalControlGroupDirective.propDecorators = {
    status: [{ type: Input }],
    layout: [{ type: Input }],
    controlAlign: [{ type: Input }],
    disabled: [{ type: Input }],
    controlWidth: [{ type: Input }],
    responsive: [{ type: Input }],
    layoutStable: [{ type: Input }]
};
